### ISSUE TITLE
image_picker Java unit test fails to resolve Maven dependencies

### ISSUE THREAD

**Issue by amirh** [flutter/flutter#48943]
**This is causing the `flutter/plugins` tree to be red**

Seems like recently maven started failing to resolve the transitive dependencies for Robolectric: [CI log](https://api.cirrus-ci.com/v1/task/5562730560094208/logs/main.log).

I believe the interesting part from the log is:

```
Starting process 'Gradle Test Executor 1'. Working directory: /tmp/cirrus-ci-build/packages/image_picker/example/android/app Command: /usr/lib/jvm/java-8-openjdk-amd64/bin/java -Djava.awt.headless=true -Djava.security.manager=worker.org.gradle.process.internal.worker.child.BootstrapSecurityManager -Dorg.gradle.native=false -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -ea -cp /home/cirrus/.gradle/caches/4.10.2/workerMain/gradle-worker.jar worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Test Executor 1'
Successfully started process 'Gradle Test Executor 1'

io.flutter.plugins.imagepicker.ImageResizerTest STANDARD_OUT
    WARNING: No manifest file found at ./AndroidManifest.xml.
    Falling back to the Android OS resources only.
    To remove this warning, annotate your test class with @Config(manifest=Config.NONE).

io.flutter.plugins.imagepicker.ImageResizerTest > onResizeImageIfNeeded_WhenQualityIsNotNull_ShoulResize_ReturnResizedFile STANDARD_ERROR
    Downloading: org/robolectric/android-all/4.1.2_r1-robolectric-0/android-all-4.1.2_r1-robolectric-0.pom from repository sonatype at https://oss.sonatype.org/content/groups/public/
    Transferring 1K from sonatype
    Downloading: org/sonatype/oss/oss-parent/7/oss-parent-7.pom from repository sonatype at https://oss.sonatype.org/content/groups/public/
    Unable to locate resource in repository

io.flutter.plugins.imagepicker.ImageResizerTest > onResizeImageIfNeeded_WhenQualityIsNotNull_ShoulResize_ReturnResizedFile STANDARD_OUT
    [INFO] Unable to find resource 'org.sonatype.oss:oss-parent:pom:7' in repository sonatype (https://oss.sonatype.org/content/groups/public/)

io.flutter.plugins.imagepicker.ImageResizerTest > onResizeImageIfNeeded_WhenQualityIsNotNull_ShoulResize_ReturnResizedFile STANDARD_ERROR
    Downloading: org/sonatype/oss/oss-parent/7/oss-parent-7.pom from repository central at http://repo1.maven.org/maven2
    Error transferring file: Server returned HTTP response code: 501 for URL: http://repo1.maven.org/maven2/org/sonatype/oss/oss-parent/7/oss-parent-7.pom

io.flutter.plugins.imagepicker.ImageResizerTest > onResizeImageIfNeeded_WhenQualityIsNotNull_ShoulResize_ReturnResizedFile STANDARD_OUT
    [WARNING] Unable to get resource 'org.sonatype.oss:oss-parent:pom:7' from repository central (http://repo1.maven.org/maven2): Error transferring file: Server returned HTTP response code: 501 for URL: http://repo1.maven.org/maven2/org/sonatype/oss/oss-parent/7/oss-parent-7.pom

io.flutter.plugins.imagepicker.ImageResizerTest > onResizeImageIfNeeded_WhenQualityIsNotNull_ShoulResize_ReturnResizedFile FAILED
    Unable to resolve artifact: Unable to get dependency information: Unable to read the metadata file for artifact 'org.robolectric:android-all:jar': Cannot find parent: org.sonatype.oss:oss-parent for project: org.robolectric:android-all:jar:4.1.2_r1-robolectric-0 for project org.robolectric:android-all:jar:4.1.2_r1-robolectric-0
      org.robolectric:android-all:jar:4.1.2_r1-robolectric-0

    from the specified remote repositories:
      central (http://repo1.maven.org/maven2),
      sonatype (https://oss.sonatype.org/content/groups/public/)

    Path to dependency: 
    	1) org.apache.maven:super-pom:pom:2.0

        Caused by:
        org.apache.maven.artifact.resolver.ArtifactResolutionException: Unable to get dependency information: Unable to read the metadata file for artifact 'org.robolectric:android-all:jar': Cannot find parent: org.sonatype.oss:oss-parent for project: org.robolectric:android-all:jar:4.1.2_r1-robolectric-0 for project org.robolectric:android-all:jar:4.1.2_r1-robolectric-0
          org.robolectric:android-all:jar:4.1.2_r1-robolectric-0

        from the specified remote repositories:
          central (http://repo1.maven.org/maven2),
          sonatype (https://oss.sonatype.org/content/groups/public/)

        Path to dependency: 
        	1) org.apache.maven:super-pom:pom:2.0

            Caused by:
            org.apache.maven.artifact.metadata.ArtifactMetadataRetrievalException: Unable to read the metadata file for artifact 'org.robolectric:android-all:jar': Cannot find parent: org.sonatype.oss:oss-parent for project: org.robolectric:android-all:jar:4.1.2_r1-robolectric-0 for project org.robolectric:android-all:jar:4.1.2_r1-robolectric-0

                Caused by:
                org.apache.maven.project.ProjectBuildingException: Cannot find parent: org.sonatype.oss:oss-parent for project: org.robolectric:android-all:jar:4.1.2_r1-robolectric-0 for project org.robolectric:android-all:jar:4.1.2_r1-robolectric-0

                    Caused by:
                    org.apache.maven.project.ProjectBuildingException: POM 'org.sonatype.oss:oss-parent' not found in repository: Unable to download the artifact from any repository

                      org.sonatype.oss:oss-parent:pom:7

                    from the specified remote repositories:
                      central (http://repo1.maven.org/maven2),
                      sonatype (https://oss.sonatype.org/content/groups/public/)

                     for project org.sonatype.oss:oss-parent

                        Caused by:
                        org.apache.maven.artifact.resolver.ArtifactNotFoundException: Unable to download the artifact from any repository

                          org.sonatype.oss:oss-parent:pom:7

                        from the specified remote repositories:
                          central (http://repo1.maven.org/maven2),
                          sonatype (https://oss.sonatype.org/content/groups/public/)

                            Caused by:
                            org.apache.maven.wagon.ResourceDoesNotExistException: Unable to download the artifact from any repository
```

Seems like it's failing to access http://repo1.maven.org/maven2/org/sonatype/oss/oss-parent/7/oss-parent-7.pom the error message for that URL says you should use https instead.

I have no idea what's going on, maybe @blasten will have a hint/

FWIW This SO post seems to discuss a similar failure: https://stackoverflow.com/questions/29939968/run-android-test-with-robolectric-dependency-error IIUC in their case it was related to some proxy configuration. I wonder whether our CI machines starting proxying or doing something with outgoing https requests? (@fkorotkov do you of any thing related that have changed recently for the Cirrus machines?)

I'm marking this as customer blocker as it's blocking all work on `flutter/plugins`.

**Comment by amirh** [comment]
It reproduces locally for me, so shouldn't be related to a Cirrus configuration change.

**Comment by dnfield** [comment]
Maven has changed their access policies as of today: https://support.sonatype.com/hc/en-us/articles/360041287334

**Comment by amirh** [comment]
Just bump into this as well :/ Do you know how can we override the http URL?

**Comment by volodia-chornenkyy** [comment]
@amirh I've was able to fix the same issue by overriding Robolectric Maven repository URL. 
Sonatype is not available for some reason and Maven Central moved to HTTPS starting from Jan 15, 2020. So, I've decided to use Maven Central URL but with a secure scheme.

Editing system property which is used by Robolectric RoboSettings class allowed me to use different URL.
`android {
    testOptions {
        unitTests.all {
            systemProperty 'robolectric.dependency.repo.url', 'https://repo1.maven.org/maven2'
        }
    }
}`
